File: plugin/src/main/java/spp/jetbrains/sourcemarker/PluginIcons.java
package spp.jetbrains.sourcemarker;

import com.intellij.openapi.util.IconLoader;

import javax.swing.*;

public interface PluginIcons {
    Icon expand = IconLoader.getIcon("/icons/expand.svg", PluginIcons.class);
    Icon expandHovered = IconLoader.getIcon("/icons/expandHovered.svg", PluginIcons.class);
    Icon expandPressed = IconLoader.getIcon("/icons/expandPressed.svg", PluginIcons.class);

    Icon close = IconLoader.getIcon("/icons/closeIcon.svg", PluginIcons.class);
    Icon closeHovered = IconLoader.getIcon("/icons/closeIconHovered.svg", PluginIcons.class);
    Icon closePressed = IconLoader.getIcon("/icons/closeIconPressed.svg", PluginIcons.class);

    Icon clock = IconLoader.getIcon("/icons/clock.svg", PluginIcons.class);
    Icon alignLeft = IconLoader.getIcon("/icons/align-left.svg", PluginIcons.class);
    Icon angleDown = IconLoader.getIcon("/icons/angle-down.svg", PluginIcons.class);

    Icon eye = IconLoader.getIcon("/icons/eye.svg", PluginIcons.class);
    Icon eyeSlash = IconLoader.getIcon("/icons/eye-slash.svg", PluginIcons.class);

    Icon analytics = IconLoader.getIcon("/icons/analytics.svg", PluginIcons.class);

    Icon config = IconLoader.getIcon("/icons/configIcon.svg", PluginIcons.class);
    Icon configHovered = IconLoader.getIcon("/icons/configIconHovered.svg", PluginIcons.class);
    Icon configPressed = IconLoader.getIcon("/icons/configIconPressed.svg", PluginIcons.class);

    Icon exclamationTriangle = IconLoader.getIcon("/icons/exclamation-triangle.svg", PluginIcons.class);
    Icon performanceRamp = IconLoader.getIcon("/icons/sort-amount-up.svg", PluginIcons.class);
    Icon activeException = IconLoader.getIcon("/icons/map-marker-exclamation.svg", PluginIcons.class);
    Icon count = IconLoader.getIcon("/icons/count.svg", PluginIcons.class);
    Icon gauge = IconLoader.getIcon("/icons/gauge.svg", PluginIcons.class);
    Icon histogram = IconLoader.getIcon("/icons/histogram.svg", PluginIcons.class);
    Icon code = IconLoader.getIcon("/icons/code.svg", PluginIcons.class);
    Icon tachometer = IconLoader.getIcon("/icons/tachometer-alt.svg", PluginIcons.class);

    interface Nodes {
        Icon variable = IconLoader.getIcon("/nodes/variable.png", PluginIcons.class);
    }

    interface Command {
        Icon logo = IconLoader.getIcon("/icons/command/logo.svg", PluginIcons.class);
        Icon clearInstrumentSelected = IconLoader.getIcon("/icons/command/clear-instruments_selected.svg", PluginIcons.class);
        Icon clearInstrumentUnSelected = IconLoader.getIcon("/icons/command/clear-instruments_unselected.svg", PluginIcons.class);

        Icon liveBreakpointSelected = IconLoader.getIcon("/icons/command/live-breakpoint_selected.svg", PluginIcons.class);
        Icon liveBreakpointUnSelected = IconLoader.getIcon("/icons/command/live-breakpoint_unselected.svg", PluginIcons.class);

        Icon livelogSelected = IconLoader.getIcon("/icons/command/live-log_selected.svg", PluginIcons.class);
        Icon livelogUnSelected = IconLoader.getIcon("/icons/command/live-log_unselected.svg", PluginIcons.class);

        Icon liveMeterSelected = IconLoader.getIcon("/icons/command/live-meter_selected.svg", PluginIcons.class);
        Icon liveMeterUnSelected = IconLoader.getIcon("/icons/command/live-meter_unselected.svg", PluginIcons.class);

        Icon liveSpanSelected = IconLoader.getIcon("/icons/command/live-span_selected.svg", PluginIcons.class);
        Icon liveSpanUnSelected = IconLoader.getIcon("/icons/command/live-span_unselected.svg", PluginIcons.class);

        Icon viewActivitySelected = IconLoader.getIcon("/icons/command/view-activity_selected.svg", PluginIcons.class);
        Icon viewActivityUnSelected = IconLoader.getIcon("/icons/command/view-activity_unselected.svg", PluginIcons.class);
        Icon viewTracesSelected = IconLoader.getIcon("/icons/command/view-traces_selected.svg", PluginIcons.class);
        Icon viewTracesUnSelected = IconLoader.getIcon("/icons/command/view-traces_unselected.svg", PluginIcons.class);
        Icon viewLogsSelected = IconLoader.getIcon("/icons/command/view-logs_selected.svg", PluginIcons.class);
        Icon viewLogsUnSelected = IconLoader.getIcon("/icons/command/view-logs_unselected.svg", PluginIcons.class);
    }

    interface Instrument {
        Icon save = IconLoader.getIcon("/icons/instrument/live-log/save.svg", PluginIcons.class);
        Icon saveHovered = IconLoader.getIcon("/icons/instrument/live-log/saveHovered.svg", PluginIcons.class);
        Icon savePressed = IconLoader.getIcon("/icons/instrument/live-log/savePressed.svg", PluginIcons.class);
    }

    interface Breakpoint {
        Icon active = IconLoader.getIcon("/icons/breakpoint/live-breakpoint-active.svg", PluginIcons.class);
        Icon complete = IconLoader.getIcon("/icons/breakpoint/live-breakpoint-complete.svg", PluginIcons.class);
        Icon disabled = IconLoader.getIcon("/icons/breakpoint/live-breakpoint-disabled.svg", PluginIcons.class);
        Icon error = IconLoader.getIcon("/icons/breakpoint/live-breakpoint-error.svg", PluginIcons.class);
        Icon pending = IconLoader.getIcon("/icons/breakpoint/live-breakpoint-pending.svg", PluginIcons.class);
    }
}

File: common/src/main/kotlin/spp/jetbrains/icons/PluginIcons.kt
/*
 * Source++, the continuous feedback platform for developers.
 * Copyright (C) 2022 CodeBrig, Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package spp.jetbrains.icons

import com.intellij.openapi.util.IconLoader

/**
 * Defines the various visual icons Source++ may display.
 *
 * @author [Brandon Fergerson](mailto:bfergerson@apache.org)
 */
object PluginIcons {
    object Nodes {
        @JvmField
        val variable = IconLoader.getIcon("/nodes/variable.png", PluginIcons::class.java)
    }

    object Command {
        @JvmField
        val logo = IconLoader.getIcon("/icons/command/logo.svg", PluginIcons::class.java)
    }

    object Instrument {
        @JvmField
        val save = IconLoader.getIcon("/icons/instrument/live-log/save.svg", PluginIcons::class.java)

        @JvmField
        val saveHovered = IconLoader.getIcon("/icons/instrument/live-log/saveHovered.svg", PluginIcons::class.java)

        @JvmField
        val savePressed = IconLoader.getIcon("/icons/instrument/live-log/savePressed.svg", PluginIcons::class.java)
    }

    object Breakpoint {
        @JvmField
        val active = IconLoader.getIcon("/icons/breakpoint/live-breakpoint-active.svg", PluginIcons::class.java)

        @JvmField
        val complete = IconLoader.getIcon("/icons/breakpoint/live-breakpoint-complete.svg", PluginIcons::class.java)

        @JvmField
        val disabled = IconLoader.getIcon("/icons/breakpoint/live-breakpoint-disabled.svg", PluginIcons::class.java)

        @JvmField
        val error = IconLoader.getIcon("/icons/breakpoint/live-breakpoint-error.svg", PluginIcons::class.java)

        @JvmField
        val pending = IconLoader.getIcon("/icons/breakpoint/live-breakpoint-pending.svg", PluginIcons::class.java)
    }

    @JvmField
    val statusEnabled = IconLoader.getIcon("/icons/statusBar/status-enabled.svg", PluginIcons::class.java)

    @JvmField
    val statusPending = IconLoader.getIcon("/icons/statusBar/status-pending.svg", PluginIcons::class.java)

    @JvmField
    val statusDisabled = IconLoader.getIcon("/icons/statusBar/status-disabled.svg", PluginIcons::class.java)

    @JvmField
    val statusFailed = IconLoader.getIcon("/icons/statusBar/status-failed.svg", PluginIcons::class.java)

    @JvmField
    val expand = IconLoader.getIcon("/icons/expand.svg", PluginIcons::class.java)

    @JvmField
    val expandHovered = IconLoader.getIcon("/icons/expandHovered.svg", PluginIcons::class.java)

    @JvmField
    val expandPressed = IconLoader.getIcon("/icons/expandPressed.svg", PluginIcons::class.java)

    @JvmField
    val close = IconLoader.getIcon("/icons/closeIcon.svg", PluginIcons::class.java)

    @JvmField
    val closeHovered = IconLoader.getIcon("/icons/closeIconHovered.svg", PluginIcons::class.java)

    @JvmField
    val closePressed = IconLoader.getIcon("/icons/closeIconPressed.svg", PluginIcons::class.java)

    @JvmField
    val clock = IconLoader.getIcon("/icons/clock.svg", PluginIcons::class.java)

    @JvmField
    val logConfig = IconLoader.getIcon("/icons/log-config.svg", PluginIcons::class.java)

    @JvmField
    val angleDown = IconLoader.getIcon("/icons/angle-down.svg", PluginIcons::class.java)

    @JvmField
    val breakpointConfig = IconLoader.getIcon("/icons/breakpoint-config.svg", PluginIcons::class.java)

    @JvmField
    val eyeSlash = IconLoader.getIcon("/icons/eye-slash.svg", PluginIcons::class.java)

    @JvmField
    val meterConfig = IconLoader.getIcon("/icons/meter-config.svg", PluginIcons::class.java)

    @JvmField
    val spanConfig = IconLoader.getIcon("/icons/span-config.svg", PluginIcons::class.java)

    @JvmField
    val count = IconLoader.getIcon("/icons/count.svg", PluginIcons::class.java)

    @JvmField
    val gauge = IconLoader.getIcon("/icons/gauge.svg", PluginIcons::class.java)

    @JvmField
    val histogram = IconLoader.getIcon("/icons/histogram.svg", PluginIcons::class.java)

    @JvmField
    val server = IconLoader.getIcon("/icons/server.svg", PluginIcons::class.java)

    @JvmField
    val chartArea = IconLoader.getIcon("/icons/chart-area.svg", PluginIcons::class.java)

    @JvmField
    val earthAmericas = IconLoader.getIcon("/icons/earth-americas.svg", PluginIcons::class.java)

    @JvmField
    val diagramSubtask = IconLoader.getIcon("/icons/diagram-subtask.svg", PluginIcons::class.java)

    @JvmField
    val squareCheck = IconLoader.getIcon("/icons/square-check.svg", PluginIcons::class.java)

    @JvmField
    val squareDashed = IconLoader.getIcon("/icons/square-dashed.svg", PluginIcons::class.java)
}
